### PR TITLE
Harden `sanitizeLog` against incorrect matches on TypeScript versions.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
# Background

Because we don't want to update every server log baseline on every TypeScript version change, we sanitize server logs based on the current TypeScript version. Specifically, we replace occurrences of the full version with `FakeVersion`, and replace occurrences of the major.minor version with `tsFakeMajor.Minor`.

# Problem

When TypeScript updating to `5.7.3`, the baselines were incorrectly updated because we have a server test with a package at the version `15.7.3`. This is not the first time we hit this issue, and historically we have gotten around it by just changing the version of the test package.

# Proposed Fix

@andrewbranch's suggestion was to switch to a regex and add boundaries to the beginning/end. This doesn't entirely fix the issue, but it does harden the test runner by avoiding a match of `5.7.3` within `15.7.3`.

### CODE REVIEW

**Review by jakebailey — APPROVED** [review]
Seems fine to me; would rather this and a cherry-pick to release-5.7 rather than manual changes.

**Review by andrewbranch — APPROVED** [review]

File: src/harness/tsserverLogger.ts [hunk 1]
```diff
@@ -111,7 +111,7 @@ export function sanitizeLog(s: string): string {
     s = s.replace(/Elapsed::?\s*\d+(?:\.\d+)?ms/g, "Elapsed:: *ms");
     s = s.replace(/"updateGraphDurationMs":\s*\d+(?:\.\d+)?/g, `"updateGraphDurationMs": *`);
     s = s.replace(/"createAutoImportProviderProgramDurationMs":\s*\d+(?:\.\d+)?/g, `"createAutoImportProviderProgramDurationMs": *`);
-    s = replaceAll(s, ts.version, "FakeVersion");
+    s = s.replace(new RegExp(`\\b${ts.version}\\b`, "g"), "FakeVersion");
```
Comment (andrewbranch) [inline src/harness/tsserverLogger.ts:114]:
```suggestion
    s = s.replace(new RegExp(`\\b${regExpEscape(ts.version)}\\b`, "g"), "FakeVersion");
```

File: src/harness/tsserverLogger.ts [hunk 2]
```diff
@@ -130,7 +130,7 @@ export function sanitizeLog(s: string): string {
     s = s.replace(/"semanticDiag":\s*\d+(?:.\d+)?/g, `"semanticDiag": *`);
     s = s.replace(/"suggestionDiag":\s*\d+(?:.\d+)?/g, `"suggestionDiag": *`);
     s = s.replace(/"regionSemanticDiag":\s*\d+(?:.\d+)?/g, `"regionSemanticDiag": *`);
-    s = replaceAll(s, `@ts${ts.versionMajorMinor}`, `@tsFakeMajor.Minor`);
+    s = s.replace(new RegExp(`\\b@ts${ts.versionMajorMinor}\\b`, "g"), `@tsFakeMajor.Minor`);
```
Comment (andrewbranch) [inline src/harness/tsserverLogger.ts:133]:
```suggestion
    s = s.replace(new RegExp(`\\b@ts${regExpEscape(ts.version)}\\b`, "g"), `@tsFakeMajor.Minor`);

```

**Review by andrewbranch — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
@typescript-bot cherry-pick to release-5.7

**Pull request merged**